### PR TITLE
[bsp/renesas]add ek-ra8d1 bsp

### PR DESCRIPTION
**Pull request by Rbb666**
## 拉取/合并请求描述：(PR description)

[
#### 为什么提交这份PR (why to submit this PR)
添加瑞萨ek-ra8d1开发板BSP

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [ ] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [ ] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [ ] 代码是高质量的 Code in this PR is of high quality
- [x] 已经使用[formatting](https://github.com/mysterywolf/formatting) 等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)

### CODE REVIEW

File: bsp/renesas/libraries/HAL_Drivers/drv_gpio.c [hunk 1]
```diff
@@ -173,7 +173,7 @@ static rt_int8_t ra_pin_read(rt_device_t dev, rt_base_t pin)
     if ((pin > RA_MAX_PIN_VALUE) || (pin < RA_MIN_PIN_VALUE))
     {
         LOG_E("GPIO pin value is illegal");
-        return -RT_ERROR;
```
Comment (mysterywolf) [inline bsp/renesas/libraries/HAL_Drivers/drv_gpio.c:176]: 这块为什么改成-1了呀？负错误码即可呀

**Review by supperthomas — CHANGES_REQUESTED** [review]
可以清理一下一些不需要的配置文件。

File: bsp/renesas/ra6m3-hmi-board/project.uvguix.rb [hunk 2]
```diff
@@ -0,0 +1,1860 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
+<ProjectGui xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="project_guix.xsd">
```
Comment (supperthomas) [inline bsp/renesas/ra6m3-hmi-board/project.uvguix.rb:2]: 这个需要吗？

**Review by supperthomas — CHANGES_REQUESTED** [review]

File: bsp/renesas/libraries/HAL_Drivers/drv_gpio.c (hunk 1 above)
Comment (Rbb666) [reply bsp/renesas/libraries/HAL_Drivers/drv_gpio.c:176]: 返回值是rt_int8_t，RT_ERROR我看改成了255啊，这不是超了吗

File: bsp/renesas/ra6m3-hmi-board/project.uvguix.rb (hunk 2 above)
Comment (Rbb666) [reply bsp/renesas/ra6m3-hmi-board/project.uvguix.rb:2]: 我删下

File: bsp/renesas/libraries/HAL_Drivers/drv_gpio.c (hunk 1 above)
Comment (mysterywolf) [reply bsp/renesas/libraries/HAL_Drivers/drv_gpio.c:176]: 要命😓

**Review by supperthomas — APPROVED** [review]

**Review by Guozhanxin — APPROVED** [review]

**Pull request merged**